src/main/java/dev/corestone/mapprotect/regions/regionmanagers/player_managers/ExplosionDamageHandler.java

```diff
@@ -63,17 +66,28 @@ public void explosionDamageListenerBlock(BlockExplodeEvent event) {
     }
 
     @EventHandler
-    public void explosionDamageListener(EntityDamageEvent event){
+    public void explosionDamageByEntityListener(EntityDamageByEntityEvent event){
         if (canExplode)return;
         if (region.getState() == RegionState.IDLE)return;
-        if (event.getCause() == EntityDamageEvent.DamageCause.BLOCK_EXPLOSION || event.getCause() == EntityDamageEvent.DamageCause.ENTITY_EXPLOSION) {
+        if (event.getCause() == EntityDamageEvent.DamageCause.ENTITY_EXPLOSION && region.getBox().contains(event.getDamager().getLocation().toVector())){
+            event.setCancelled(true);
```

Comment:
There is a difference between damager and the damaged. Here, you are seeing if the damager is inside of the box. We need to check both. event.getEntity() gets the damaged entity and event.getDamager() gets the entity that exploded.